src/utils.cpp

```diff
@@ -1104,6 +1104,20 @@ FuncsModulesMap parse_traceable_funcs()
                                       ? path_env
                                       : tracefs::available_filter_functions();
 
+  /*
+   * Certain kernel functions are known to cause system stability issues if
+   * traced yet they are not specified in the blacklist so are filtered out
+   * as the list is built. The list of functions have been taken from
+   * the bpf kernel selftests (bpf/prog_tests/kprobe_multi_test.c).
+   */
+  const std::unordered_set<std::string> bad_funcs = {
+    "arch_cpu_idle", "default_idle", "bpf_dispatcher_xdp_func"
+  };
+
+  const std::vector<std::string> bad_funcs_partial = {
+    "__ftrace_invalid_address__", "rcu_"
+  };
```

Comment:
Can we move these closer to https://github.com/bpftrace/bpftrace/blob/c532239f4b42eda260fcc8aa9ece56fecc1d083e/src/attached_probe.cpp#L36-L45 ? Would be good to keep all these together in the same location. Doesn't necessarily have to be in attached_probe.cpp

Reply:
> Can we move these closer to
> 
> https://github.com/bpftrace/bpftrace/blob/c532239f4b42eda260fcc8aa9ece56fecc1d083e/src/attached_probe.cpp#L36-L45
> 
> ? Would be good to keep all these together in the same location. Doesn't necessarily have to be in attached_probe.cpp

We can have a chat tomorrow but that mechanism doesn't seem to be a good one to me. Why not just remove the functions from the ones that are available instead of having this mechanism which is more intrusive and provides no extra benefit.

Reply:
> > Can we move these closer to
> > https://github.com/bpftrace/bpftrace/blob/c532239f4b42eda260fcc8aa9ece56fecc1d083e/src/attached_probe.cpp#L36-L45
> > 
> > ? Would be good to keep all these together in the same location. Doesn't necessarily have to be in attached_probe.cpp
> 
> We can have a chat tomorrow but that mechanism doesn't seem to be a good one to me. Why not just remove the functions from the ones that are available instead of having this mechanism which is more intrusive and provides no extra benefit.

Just a follow on: the `banned_kretprobes` mechanism is most likely being removed in #3206 .

Reply:
There is still going to be a list of functions but I can move those into utils.cpp to keep them with these.

Reply:
> There is still going to be a list of functions but I can move those into utils.cpp to keep them with these.

Up to you @jordalgo but probably makes sense to keep them together for ease of maintenance.

Reply:
`utils.cpp` is already long enough. Maybe it'd be worth having a separate source file for these, to simplify maintenance.